### PR TITLE
Feature/policy values

### PR DESCRIPTION
**Pull request by dpetran**
Closes https://github.com/fluree/core/issues/128

This change makes the `opts.policyValues` use the same syntax as an fql values clause or a values pattern.
```
["?a" [1 2]]
;; or
[["?a "?b] [["a1" "b1"] ["a2" "b2"]]]
```
It composes with an existing values clause on the policy query by adding the specified values as a values pattern in the where clause.

### CODE REVIEW

File: src/clj/fluree/db/query/fql/parse.cljc [hunk 1]
```diff
@@ -145,19 +146,15 @@
...
-      (if (every? (fn [binding]
-                    (= (count binding) var-count))
-                  vals*)
-        [vars* (mapv (fn [vals**]
-                       (parse-value-binding vars* vals** context))
-                     vals*)]
-        (throw (ex-info (str "Invalid value binding: "
-                             "number of variables and values don't match: "
-                             values)
+    (let [[vars vals] (policy/normalize-values values)
```
Comment (zonotope) [inline src/clj/fluree/db/query/fql/parse.cljc:149]: If that function will be used for more than just policy related features then it should live in a different namespace.

File: test/fluree/db/policy/query_test.clj [hunk 2]
```diff
@@ -293,3 +290,95 @@
...
+                                            "ex:capitol" "Charlotte"}
+                                           {"@id" "usa:co"
+                                            "@type" "usa:state"
+                                            "ex:name" "Colorado"
+                                            "ex:capitol" "Denver"}
+                                           {"@id" "usa:pr"
+                                            "@type" "usa:territory"
+                                            "ex:name" "Puerto Rico"
+                                            "ex:capitol" "San Juan"}]})]
+    (testing "no policyValues"
```
Comment (zonotope) [inline test/fluree/db/policy/query_test.clj:319]: What's supposed to happen when there are no policyValues? This test needs more commentary

**Review by zonotope — APPROVED** [review]
I have two changes that I pointed out, but this looks good otherwise.

File: src/clj/fluree/db/query/fql/parse.cljc (hunk 1 above)
Comment (dpetran) [reply src/clj/fluree/db/query/fql/parse.cljc:149]: I had it in `parse` originally, but that unfortunately introduced a massive cyclic dependency. I'll see if I can untangle it another way, I agree that this is not the right home for it.

Comment (dpetran) [reply src/clj/fluree/db/query/fql/parse.cljc]: I ended up putting it in a new `util` ns, I couldn't really find another place that seemed correct and this seems like the least bad way.

**Pull request merged**